File: library/src/main/java/com/pengrad/telegrambot/model/request/InlineQueryResultDocument.java
package com.pengrad.telegrambot.model.request;

import java.io.Serializable;

/**
 * Stas Parshin
 * 06 May 2016
 */
public class InlineQueryResultDocument extends InlineQueryResult<InlineQueryResultDocument> implements Serializable {
    private final static long serialVersionUID = 0L;

    public static final String MIME_APP_PDFL = "application/pdf";
    public static final String MIME_APP_ZIP = "application/zip";

    private String title;
    private String document_url;
    private String mime_type;

    private String caption;
    private String parse_mode;
    private String description;

    private String thumbnail_url;
    private Integer thumbnail_width;
    private Integer thumbnail_height;


    public InlineQueryResultDocument(String id, String documentUrl, String title, String mimeType) {
        super("document", id);
        this.document_url = documentUrl;
        this.title = title;
        this.mime_type = mimeType;
    }

    public InlineQueryResultDocument caption(String caption) {
        this.caption = caption;
        return this;
    }

    public InlineQueryResultDocument parseMode(ParseMode parseMode) {
        this.parse_mode = parseMode.name();
        return this;
    }

    public InlineQueryResultDocument description(String description) {
        this.description = description;
        return this;
    }

    /**
     * @deprecated Use thumbnailUrl instead
     */
    @Deprecated
    public InlineQueryResultDocument thumbUrl(String thumbUrl) {
        return thumbnailUrl(thumbUrl);
    }

    /**
     * @deprecated Use thumbnailWidth instead
     */
    @Deprecated
    public InlineQueryResultDocument thumbWidth(Integer thumbWidth) {
        return thumbnailWidth(thumbWidth);
    }

    /**
     * @deprecated Use thumbnailHeight instead
     */
    @Deprecated
    public InlineQueryResultDocument thumbHeight(Integer thumbHeight) {
        return thumbnailHeight(thumbHeight);
    }


    public InlineQueryResultDocument thumbnailUrl(String thumbnailUrl) {
        this.thumbnail_url = thumbnailUrl;
        return this;
    }

    public InlineQueryResultDocument thumbnailWidth(Integer thumbnailWidth) {
        this.thumbnail_width = thumbnailWidth;
        return this;
    }

    public InlineQueryResultDocument thumbnailHeight(Integer thumbnailHeight) {
        this.thumbnail_height = thumbnailHeight;
        return this;
    }
}

File: core/src/commonMain/kotlin/com/pengrad/telegrambot/model/request/InlineQueryResultDocument.kt
package com.pengrad.telegrambot.model.request

import java.io.Serializable

/**
 * Stas Parshin
 * 06 May 2016
 */
class InlineQueryResultDocument(id: String, private val document_url: String, private val title: String, private val mime_type: String) : InlineQueryResult<InlineQueryResultDocument>("document", id), Serializable {

    private var caption: String? = null
    private var parse_mode: String? = null
    private var description: String? = null
    private var thumb_url: String? = null
    private var thumb_width: Int? = null
    private var thumb_height: Int? = null

    fun caption(caption: String): InlineQueryResultDocument {
        this.caption = caption
        return this
    }

    fun parseMode(parseMode: ParseMode): InlineQueryResultDocument {
        this.parse_mode = parseMode.name
        return this
    }

    fun description(description: String): InlineQueryResultDocument {
        this.description = description
        return this
    }

    fun thumbUrl(thumbUrl: String): InlineQueryResultDocument {
        this.thumb_url = thumbUrl
        return this
    }

    fun thumbWidth(thumbWidth: Int?): InlineQueryResultDocument {
        this.thumb_width = thumbWidth
        return this
    }

    fun thumbHeight(thumbHeight: Int?): InlineQueryResultDocument {
        this.thumb_height = thumbHeight
        return this
    }

    companion object {
        private const val serialVersionUID = 0L

        val MIME_APP_PDFL = "application/pdf"
        val MIME_APP_ZIP = "application/zip"
    }
}
